src/IO/DataImporter/DataFileReaderActions.hpp

```diff
@@ -0,0 +1,219 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <cstddef>
+#include <iterator>
+#include <unordered_map>
+
+#include "DataStructures/DataBox/DataBox.hpp"
+#include "DataStructures/DataBox/DataBoxTag.hpp"
+#include "DataStructures/Index.hpp"
+#include "DataStructures/Tensor/TensorData.hpp"
+#include "ErrorHandling/Error.hpp"
+#include "IO/DataImporter/Tags.hpp"
+#include "IO/H5/AccessType.hpp"
+#include "IO/H5/File.hpp"
+#include "IO/H5/VolumeData.hpp"
+#include "IO/Observer/ArrayComponentId.hpp"
+#include "Parallel/ConstGlobalCache.hpp"
+#include "Parallel/Info.hpp"
+#include "Parallel/Invoke.hpp"
+#include "Utilities/Algorithm.hpp"
+#include "Utilities/Requires.hpp"
+#include "Utilities/TaggedTuple.hpp"
+
+namespace importer {
+/// Actions related to the data importer
+namespace Actions {
+
+/*!
+ * \brief Invoked on the `importer::DataFileReader` component to store the
+ * registered data.
+ *
+ * The `importer::Actions::RegisterWithImporter` action, which is performed on
+ * each element of an array parallel component, invokes this action on the
+ * `importer::DataFileReader` component.
+ */
+struct RegisterElementWithSelf {
+  template <
```

Comment:
Is this spacing the output of clang-format?

Reply:
yes, this is clang-formatted.